### PR TITLE
testing improvements on layout

### PR DESCRIPTION
**Pull request by farfromrefug**
Related to this issue https://github.com/NativeScript/NativeScript/issues/7988

For now this PR only checks for auto in width or height. This is clearly not enough.
But the idea is to check if it is necessary to requestLayout on the parent. Most of the times it is not necessary.

In the case of a complex app the improvements is huge!. It makes apps using animations (not using {N} animation system which is too limited) far more snappier.

This is only a POC!

### CODE REVIEW

**Comment by vtrifonov** [comment]
test

**Comment by vtrifonov** [comment]
You needed to run `npm run api-extractor` in the root folder which will regenerate the NaitiveScript.api.md file as there is a change in the public API

**Comment by farfromrefug** [comment]
@vtrifonov will try and do that as soon as possible

**Comment by farfromrefug** [comment]
@vtrifonov just pushed the update

File: nativescript-core/ui/core/view-base/view-base.d.ts [hunk 1]
```diff
@@ -310,7 +310,7 @@ export abstract class ViewBase extends Observable {
     /**
      * Invalidates the layout of the view and triggers a new layout pass.
      */
-    public requestLayout(): void;
+    public requestLayout(calledFromParent?:boolean): void;
```
Comment (vtrifonov) [inline nativescript-core/ui/core/view-base/view-base.d.ts]: The same here, maybe the parameter name should be **calledFromChild**
```suggestion
    public requestLayout(calledFromChild?:boolean): void;
```

File: nativescript-core/ui/core/view/view-common.ts [hunk 2]
```diff
@@ -819,9 +819,9 @@ export abstract class ViewCommon extends ViewBase implements ViewDefinition {
         }
     }
 
-    public requestLayout(): void {
+    public requestLayout(calledFromChild): void {
```
Comment (vtrifonov) [inline nativescript-core/ui/core/view/view-common.ts]: it should have a default value here as in the view-base
```suggestion
    public requestLayout(calledFromChild = false): void {
```

File: api-reports/NativeScript.api.md [hunk 3]
```diff
@@ -2915,7 +2915,7 @@ export abstract class ViewBase extends Observable {
     public _removeViewCore(view: ViewBase): void;
     // (undocumented)
     _removeViewFromNativeVisualTree(view: ViewBase): void;
-    public requestLayout(): void;
+    public requestLayout(calledFromParent?:boolean): void;
```
Comment (vtrifonov) [inline api-reports/NativeScript.api.md]: I guess this should be **calledFromChild** (as it is in the view-base.ts and view-common.ts files) but not **calledFromParent**
```suggestion
    public requestLayout(calledFromChild?:boolean): void;
```

Comment (farfromrefug) [reply api-reports/NativeScript.api.md]: @vtrifonov actually already changed that here. Need to pus hit!

**Comment by farfromrefug** [comment]
@vtrifonov as discussed with @vakrilov i will close that PR and open a new one as Draft PR.
This feature is a POC and needs a bit more work. We need to discuss this

**Pull request closed without merging**